pioemu/instructions/push.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2023 Aaronjamt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from dataclasses import replace
+from pioemu.state import ShiftRegister
+from pioemu.conditions import receive_fifo_not_full
+
+
+def push_blocking(state):
+    state.receive_fifo.push(state.input_shift_register.contents)
```

Comment:
That's weird, I don't have the same issue. What version of Python are you using? I'm using 3.11.0rc2 and it works just fine. `push()` will add the value to the beginning of the FIFO and `append()` adds to the end so I believe it should be `push()` not `append()`

Reply:
2. Although `deque` provides both `appendleft` and `insert` methods I believe `append` is the more appropriate choice. I will give an example. If you were using `pioemu` to test / debug a serial UART then you would logically expected the `receive_fifo` to containing the values _received_ in chronological order e.g. [1, 2, 3, 4]. If the latest value was inserted at the start of the `deque` then the FIFO's values would be reversed. This does means that any test / debug code using `pioemu` would need to use `popleft()`  instead of `pop()`. I need to document some of the design decisions taken with this library so I'll take on the task of documenting this expectation.

Reply:
I see, that makes sense. I'll change it to just `append()`